manifests/extract.pp

```diff
@@ -41,25 +41,30 @@
   $strip_components=0,
   $purge=false,
   $user=undef,
-  $tar_command=$::osfamily ? {
-    'Solaris' => 'gtar',
-    default   => 'tar',
-  },
+  $tar_command=undef,
 ) {
 
+  require ::archive::params
```

Comment:
I'm not exactly sure about that one. Does an include enforce order of evaluation?

the require ensures ::archive::params is evaluated before the extract.pp is evaluated, so that the
values of ::archive::params::FOOBAR are available, when used later in the defined type, does the
include enforce the same?